test/internet/test-uv-threadpool-schedule.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+
+// test to validate massive dns lookups do not block filesytem I/O
+// (or any fast I/O).Prior to https://github.com/libuv/libuv/pull/1845
+// few back-to-back dns lookups were sufficient to engage libuv
+// threadpool workers in a blocking manner, throttling other work items
+// that need pool resources. Start slow and fast I/Os together, make sure
+// fast I/O can complete in at least in 1/100th of time for slow I/O.
+// TEST TIME TO COMPLETION: ~5 seconds.
+
+const common = require('../common');
+const dns = require('dns');
+const fs = require('fs');
+const assert = require('assert');
+
+const start = Date.now();
+
+const slowIOmax = 100;
+let slowIOcount = 0;
+let fastIOdone = false;
+let slowIOend, fastIOend;
+const randomDomain = 'nonexistent';
+
+for (let i = 0; i < slowIOmax; i++) {
+  // We need to refresh the domain string everytime,
+  // otherwise the TCP stack that cache the previous lookup
+  // returns result from memory, breaking all our Math.
+  dns.lookup(`${randomDomain}${i}.com`, {}, common.mustCall(() => {
```

Comment:
Does the test still work if we use one of the reserved TLDs (e.g. `.test`)? https://tools.ietf.org/html/rfc2606

Reply:
```patch
-  dns.lookup(`${randomDomain}${i}.com`, {}, common.mustCall(() => {
+  dns.lookup(`${randomDomain}${i}.test`, {}, common.mustCall(() => {
```

@richardlau - I tested with this change (if that is what you meant) and it still works. But what would be an advantage of it?

Reply:
The reserved TLDs are specifically reserved to avoid conflicts with actual registered domain names, e.g. `nonexistent0.com` might exist in the future. 

They can also avoid unnecessary load on the DNS servers, see https://tools.ietf.org/html/rfc6761 section 6.

Reply:
@richardlau : ok - let me clarify: the purpose of the test is to engage (half of) the libuv threadpool workers as long as possible while making sure at least one thread is free and available to serve the file I/Os.  So in that context, what we want is a list of domain names that are unique, and were unresolved earlier (resolved names are cached so retrieved faster than we wanted them to), existent or non-existent is not a consideration for the test. If the name is confusing, I can change it to `unique0.com` etc. 

Am I missing anything?

Reply:
If caching is an issue, we might need to generate domain names that are unique per-test-run, e.g. based on `Date.now()` or `Math.random()`?

Reply:
yes (my first commit 1d0fcefec8a42b71166e572de14b308e1dc97e58) had it based on Math.random(). I guess I will re-instate that, thanks.